### PR TITLE
feat: auth repository

### PR DESCRIPTION
**Pull request by hchaviano**

### CODE REVIEW

File: packages/auth/lib/src/auth_service.dart [hunk 1]
```diff
@@ -25,7 +23,8 @@ abstract class AuthService {
 
   Future<void> sendPasswordResetEmail({@required String email});
 
-  Future<User> signInWithSocialMedia({@required SocialMediaMethod method});
+  Future<UserEntity> signInWithSocialMedia(
+      {@required SocialMediaMethod method});
```
Comment (gonzasosa) [inline packages/auth/lib/src/auth_service.dart:27]:
```suggestion
      {@required SocialMediaMethod method,});
```

File: packages/auth/lib/src/firebase_init_service.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    socialMediasAuth.forEach((element) {
+      if (_services.containsKey(element)) {
+        _serviceFactory.addService(
+            method: element, constructor: () => _services[element]);
+      }
+    });
+
+    return FirebaseAuthService(
+        authService: FirebaseAuth.instance,
+        signInServiceFactory: _serviceFactory);
```
Comment (gonzasosa) [inline packages/auth/lib/src/firebase_init_service.dart:27]:
```suggestion
        signInServiceFactory: _serviceFactory,);
```

File: packages/auth/lib/src/firebase_init_service.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ),
+  };
+
+  FirebaseAuthService init(List<SocialMediaMethod> socialMediasAuth) {
+    final _serviceFactory = SignInServiceFactory();
+
+    socialMediasAuth.forEach((element) {
+      if (_services.containsKey(element)) {
+        _serviceFactory.addService(
+            method: element, constructor: () => _services[element]);
```
Comment (gonzasosa) [inline packages/auth/lib/src/firebase_init_service.dart:21]:
```suggestion
            method: element, constructor: () => _services[element],);
```

File: packages/auth/lib/src/firebase_init_service.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
+part of auth;
+
+class FirebaseInitService {
+  final Map<SocialMediaMethod, ISignInService> _services = {
+    SocialMediaMethod.APPLE:
+        AppleSignInService(appleCredentials: const AppleCredentials()),
```
Comment (gonzasosa) [inline packages/auth/lib/src/firebase_init_service.dart:6]:
```suggestion
        AppleSignInService(appleCredentials: const AppleCredentials(),),
```

File: test/services/auth/sign_in_services/google_sign_in_service_test.dart [hunk 5]
```diff
@@ -1,8 +1,8 @@
-import 'package:firebase_auth/firebase_auth.dart';
-import 'package:firebasestarter/services/auth/auth.dart';
+import 'package:auth/auth.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:google_sign_in/google_sign_in.dart';
 import 'package:mockito/mockito.dart';
+import 'package:firebase_auth/firebase_auth.dart' as auth;
```
Comment (gonzasosa) [inline test/services/auth/sign_in_services/google_sign_in_service_test.dart:5]:
```suggestion
import 'package:firebase_auth/firebase_auth.dart' as Auth;
```

File: lib/edit_profile/bloc/edit_profile_bloc.dart [hunk 6]
```diff
@@ -193,7 +193,7 @@ class EditProfileBloc extends Bloc<EditProfileEvent, EditProfileState> {
       }
 
       if (needToUpdateImage || needToUpdateDisplayName) {
-        return await _authService.currentUser();
+        return _toUser(await _authService.currentUser());
```
Comment (gonzasosa) [inline lib/edit_profile/bloc/edit_profile_bloc.dart:196]:
```suggestion
        final user = await _authService.currentUser();
        return _toUser(user);
```

File: lib/models/user.dart [hunk 7]
```diff
@@ -6,21 +7,19 @@ class User extends Equatable {
   final String lastName;
   final String email;
   final String imageUrl;
-  final bool isAnonymous;
-  final int age;
-  final String phoneNumber;
-  final String address;
+  // final int age;
+  // final String phoneNumber;
+  // final String address;
```
Comment (gonzasosa) [inline lib/models/user.dart:12]: Remove this comments

File: lib/models/user.dart [hunk 8]
```diff
@@ -6,21 +7,19 @@ class User extends Equatable {
...
     this.lastName,
     this.email,
     this.imageUrl,
-    this.isAnonymous,
-    this.age,
-    this.phoneNumber,
-    this.address,
+    // this.age,
+    // this.phoneNumber,
+    // this.address,
```
Comment (gonzasosa) [inline lib/models/user.dart:22]: Please also remove this comments

File: lib/models/user.dart [hunk 9]
```diff
@@ -29,32 +28,41 @@ class User extends Equatable {
         lastName: json['lastName'],
         email: json['email'],
         imageUrl: json['imageUrl'],
-        isAnonymous: json['isAnonymous'],
-        age: json['age'],
-        phoneNumber: json['phoneNumber'],
-        address: json['address'],
+        // age: json['age'],
+        // phoneNumber: json['phoneNumber'],
+        // address: json['address'],
```
Comment (gonzasosa) [inline lib/models/user.dart:33]: Same here about the comments

File: lib/models/user.dart [hunk 10]
```diff
@@ -29,32 +28,41 @@ class User extends Equatable {
...
   Map<String, dynamic> toJson() => {
         'firstName': firstName,
         'lastName': lastName,
         'email': email,
-        'age': age,
-        'address': address,
-        'phoneNumber': phoneNumber,
+        // 'age': age,
+        // 'address': address,
+        // 'phoneNumber': phoneNumber,
```
Comment (gonzasosa) [inline lib/models/user.dart:42]: Same on comments

File: lib/models/user.dart [hunk 11]
```diff
@@ -29,32 +28,41 @@ class User extends Equatable {
...
       };
 
+  static User fromEntity(UserEntity entity) => User(
+        id: entity.id,
+        firstName: entity.firstName,
+        lastName: entity.lastName,
+        email: entity.email,
+        // age: entity.age,
+        // phoneNumber: entity.phoneNumber,
+        // address: entity.address,
```
Comment (gonzasosa) [inline lib/models/user.dart:52]: Same on comments

File: lib/models/user.dart [hunk 12]
```diff
@@ -29,32 +28,41 @@ class User extends Equatable {
...
         lastName,
         email,
         imageUrl,
-        isAnonymous,
-        age,
-        phoneNumber,
-        address
+        // age,
+        // phoneNumber,
+        // address
```
Comment (gonzasosa) [inline lib/models/user.dart:65]: Also remove this one 😅

**Review by gonzasosa — CHANGES_REQUESTED** [review]

File: test/services/auth/sign_in_services/google_sign_in_service_test.dart (hunk 5 above)
Comment (hchaviano) [reply test/services/auth/sign_in_services/google_sign_in_service_test.dart:5]: The new linter recommends using lowerCamelCase for this case

**Review by gonzasosa — APPROVED** [review]

**Pull request merged**